### ISSUE TITLE
Question: why re-execute binary on each query?

### ISSUE THREAD

**Issue by linonetwo** [cocktailpeanut/dalai#95]
Hi, I'm learning your code, I'm curious that you `exec` the `alpaca.cpp`'s binary on each query, am I right?

https://github.com/cocktailpeanut/dalai/blob/b8a7af8262a3e8d9082a3596eadf8bdd307b4fe7/index.js#L202-L211

Will this cause re-initialization that will load and unload memory that consumes some time?

**Comment by ikcikoR** [comment]
Yeah I kinda hate it, my HDD isn't the fastest so this is a bunch of downtime with every prompt...

**Comment by linonetwo** [comment]
Oh, I thought dalai don't have this problem... I didn't get dalai download the model, so I download the modal manually from https://github.com/facebookresearch/llama/pull/73 and use llama.cpp instead.

But I still want to solve this.

**Comment by linonetwo** [comment]
https://github.com/linonetwo/langchain-alpaca/blob/1c519c206ea5152269c3d82d160a2567bbc9e69f/src/index.ts#L222-L282

I have to do it the same way, until https://github.com/antimatter15/alpaca.cpp/issues/86 solved
